src/models/Level.php

```diff
@@ -654,106 +654,156 @@ private static function levelFromRow(Map<string, string> $row): Level {
     );
   }
 
+  private static async function withLock(
+    string $lock_name,
+    int $team_id,
+    (function(): Awaitable<mixed>) $thunk,
+  ): Awaitable<mixed> {
+    $lock_name = sprintf('%s/%s_%d', sys_get_temp_dir(), $lock_name, $team_id);
```

Comment:
Fair enough. That is a good solution. Push push! ![Approved] (http://a.fod4.com/misc/rambo%20original.gif)